### PR TITLE
added acs-engine code delivery guidelines document

### PR DESCRIPTION
**Pull request by dmitsh**
added acs-engine code delivery guidelines

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/970)

### CODE REVIEW

**Comment by seanknox** [comment]
@dmitsh FWIW using `[skip ci]` in a commit message is a pattern lots of devs are familiar with–I think this will help!

File: docs/acsengine-code-delivery.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# ACS Engine code delivery guide
+
+[![ACS Engine](https://azurecomcdn.azureedge.net/mediahandler/acomblog/media/Default/blog/a8f28783-3ddc-4081-a57d-6d97147467bf.png)](https://github.com/azure/acs-engine)
+
+ACS Engine is an open source project to generate ARM (Azure Resource Manager) templates DC/OS, Kubernetes, Swarm Mode clusters on Microsoft Azure.
+ACS Engine code lives in Azure, and it is imperative to ensure its quality.
+This document provides guidelines to achieve that.
+
```
Comment (seanknox) [inline docs/acsengine-code-delivery.md]: I'd replace:

> ACS Engine code lives in Azure, and it is imperative to ensure its quality.
> This document provides guidelines to achieve that.

With something like:

> This documents provides guidelines to the acs-engine testing and continuous integration process.

File: docs/acsengine-code-delivery.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  + generates multiple ARM templates for different deployment scenarios
+  + simultaneously provisions the clusters based on generated templates in Azure
+  This test might take 20-40 minutes.
+  If the test fails, review the logs. If the failure was caused by your code change, return to step (1).
+  Sometimes the test might fail because of intermittent Azure issues, such as resource unavailability of provisioning timeout. In this case manually trigger Jenkins PR job from your GitHub PR page.
+> **Note:** You may want to suppress triggering the PR builds, for example, when you add/change the documentation, or when you are in the middle of your changes but want to make them visible to others.
+> There are few ways to do so:
+> (1) applying ***in progress*** or ***kind/documentation*** label to your PR
+> (2) adding ***[skip ci]*** to the PR title
+> Make sure to mark your PR prior to submitting it.
```
Comment (seanknox) [inline docs/acsengine-code-delivery.md]: Is `[skip ci]` now functional? Hooray if so.

File: docs/acsengine-code-delivery.md (hunk 1 above)
Comment (dmitsh) [inline docs/acsengine-code-delivery.md]: done

File: docs/acsengine-code-delivery.md (hunk 2 above)
Comment (dmitsh) [inline docs/acsengine-code-delivery.md]: [skip ci] is functional in the PR title. Doesn't work in the checking comment.

Comment (seanknox) [inline docs/acsengine-code-delivery.md]: OK, skipping entire PRs isn't super useful, unfortunately. We need to be able to skip by commit. I should have been clearer with the acceptance criteria, sorry.

**Comment by dmitsh** [comment]
@seanknox please review

**Comment by seanknox** [comment]
@acs-bot test this please

**Pull request merged**